staging/src/k8s.io/apiserver/pkg/storage/cacher/watch_progress.go

```diff
@@ -63,6 +67,7 @@ type conditionalProgressRequester struct {
 
 func (pr *conditionalProgressRequester) Run(stopCh <-chan struct{}) {
 	ctx := wait.ContextForChannel(stopCh)
+	ctx = metadata.NewOutgoingContext(ctx, pr.contextMetadata)
```

Comment:
make this context wrapping conditional on whether pr.contextMetadata is nil or not?